src/app/[lang]/app/search-pro/page.tsx

```diff
@@ -0,0 +1,182 @@
+'use client';
+import { useState, type FC } from 'react';
+import { useTransition, animated } from '@react-spring/web';
+import clsx from 'clsx';
+// components
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { Checkbox } from '@/modules/core/components/checkbox';
+import { DateSelect } from '@/modules/core/components/date-select';
+import { Button } from '@/modules/core/components/button';
+import { ProSearchField } from '@/modules/location/components/pro-search-field';
+// hooks
+import { useBoolean } from 'usehooks-ts';
+// containers
+import { ProfessionalSearchCard } from '@/modules/user/containers/professional-search-card';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+// styles
+import styles from './search-pro.module.scss';
+import type { ChildrenProp } from '@/modules/core/types/react.types';
+import { useDeviceType } from '@/modules/core/hooks/use-device-type';
+import { DialogFullScreen } from '@/modules/core/components/dialog-full-screen';
+import { Footer } from '@/modules/landing/containers/footer/footer';
+
+const FilterWrapper: FC<
+  ChildrenProp & {
+    isActive: boolean;
+    onActiveChange: (isActive: boolean) => void;
+  }
+> = ({ children, isActive, onActiveChange }) => {
+  const deviceType = useDeviceType();
+  const transitions = useTransition(isActive, {
+    from: { transform: 'translateX(100%)', opacity: 0 },
+    enter: { transform: 'translateX(0%)', opacity: 1 },
+    leave: { transform: 'translateX(100%)', opacity: 0 },
+    config: { tension: 330, friction: 30 },
+  });
+
+  if (deviceType === 'mobile' || deviceType === 'tablet') {
+    return (
+      <DialogFullScreen
+        isOpen={isActive}
+        onOpenChange={onActiveChange}
+        closeOnOutsideClick
+        classes={{
+          content: 'max-w-[300px]',
+        }}
+      >
+        {children}
+      </DialogFullScreen>
+    );
+  }
+
+  return (
+    <>
+      {transitions(
+        (style, item) =>
+          item && (
+            <animated.div
+              style={style}
+              className='w-[300px] rounded-lg h-fit bg-white shadow sticky right-0 md:top-[150px] top-[120px]'
+            >
+              {children}
+            </animated.div>
+          )
+      )}
+    </>
+  );
+};
+
+export default function SearchProPage() {
+  const { data: professionalList } = trpc.professional.list.useQuery({
+    limit: 6,
+  });
+  // state
+  const [date, setDate] = useState<null | Date>(null);
+
+  const isFilterActive = useBoolean(false);
+
+  const services = ['All', 'Makeup', 'Fitness', 'Hair', 'Nails', 'Massage'];
```

Comment:
Those services should come from the backend, but it can be done in another pr

Reply:
done